### PR TITLE
libusb: bump to 1.0.23

### PR DESCRIPTION
**Pull request by kenmays**
Built successfully on hrev53452.

### CODE REVIEW

File: dev-libs/libusb/patches/libusb-1.0.23.patchset [hunk 1]
```diff
@@ -0,0 +1,59 @@
+From 5aa523289f82d1c7f01dde601356214920646542 Wed Jul 31 00:00:00 2019
+From: Ken Mays <kmays2000@gmail.com>
+Date: Thu, Sep 5 2019 01:15:08 +0000
+Subject: Fix Haiku port
```
Comment (waddlesplash) [inline dev-libs/libusb/patches/libusb-1.0.23.patchset]: Were all the other patches upstreamed?

Comment (pulkomandy) [reply dev-libs/libusb/patches/libusb-1.0.23.patchset]: I upstreamed what we had in the previous version, yes.

**Comment by extrowerk** [comment]
Builds and packages fine on 64 bit.

**Comment by waddlesplash** [comment]
IIRC it won't anymore on gcc2. Do we care?

**Comment by Begasus** [comment]
Builds and package ok on gcc2h for _x86, as expected build fails for gcc2 with:
/sources/libusb-1.0.23/libusb/os/haiku_usb_raw.cpp:214: parse error before `.'

File: dev-libs/libusb/libusb-1.0.23.recipe [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	2012-2013 Toby Gray
+	2013-2018 Chris Dickens"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://github.com/libusb/libusb/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="02620708c4eea7e736240a623b0b156650c39bfa93a14bcfa5f3e05270313eba"
+SOURCE_FILENAME="libusb-$portVersion.tar.gz"
+PATCHES="libusb-$portVersion.patchset"
+
+ARCHITECTURES="x86 x86_64 !x86_gcc2"
```
Comment (Begasus) [inline dev-libs/libusb/libusb-1.0.23.recipe]: Suggested change (style):
```diff
-ARCHITECTURES="x86 x86_64 !x86_gcc2"
+ARCHITECTURES="x86_gcc2 ?x86 x86_64"
```
There is no x86 (primary architecture, so we can't check it there) :)

Comment (kenmays) [reply dev-libs/libusb/libusb-1.0.23.recipe]: Done

File: dev-libs/libusb/patches/libusb-1.0.23.patchset [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
++	}
++}
+ 
+ 
+ USBDevice::USBDevice(const char *path)
+@@ -449,6 +466,7 @@ USBDevice::EndpointToInterface(uint8 address) const
+ int
+ USBDevice::Initialise()		//Do we need more error checking, etc? How to report?
+ {
++	usbi_dbg("Haiku init dev %s", fPath);
```
Comment (waddlesplash) [inline dev-libs/libusb/patches/libusb-1.0.23.patchset]: What's the point of this?

File: dev-libs/libusb/patches/libusb-1.0.23.patchset [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+diff --git a/libusb/os/haiku_usb_backend.cpp.older b/libusb/os/haiku_usb_backend.cpp
+index 539a996..16d7528 100644
+--- a/libusb/os/haiku_usb_backend.cpp.older
++++ b/libusb/os/haiku_usb_backend.cpp
+@@ -309,7 +309,7 @@ USBDeviceHandle::SetAltSetting(int inumber, int alt)
+ 	return LIBUSB_SUCCESS;
+ }
+ 
+-
++/*
```
Comment (waddlesplash) [inline dev-libs/libusb/patches/libusb-1.0.23.patchset]: And this?

**Comment by pulkomandy** [comment]
Why are there still patches at all? The required changes to libusb had already been upstreamed

**Comment by kenmays** [comment]
@pulkomandy - Some of the haiku_usb_backend code got missed.

File: dev-libs/libusb/patches/libusb-1.0.23.patchset (hunk 4 above)
Comment (pulkomandy) [reply dev-libs/libusb/patches/libusb-1.0.23.patchset]: As far as I can see the code in upstream is there:
https://github.com/libusb/libusb/blob/master/libusb/os/haiku_usb_backend.cpp

File: dev-libs/libusb/patches/libusb-1.0.23.patchset (hunk 3 above)
Comment (pulkomandy) [reply dev-libs/libusb/patches/libusb-1.0.23.patchset]: Should be removed.

File: dev-libs/libusb/patches/libusb-1.0.23.patchset [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
++	usbi_dbg("Haiku init dev %s", fPath);
+ 	int fRawFD = open(fPath, O_RDWR | O_CLOEXEC);
+ 	if (fRawFD < 0)
+ 		return B_ERROR;
+@@ -479,7 +497,7 @@ USBDevice::Initialise()		//Do we need more error checking, etc? How to report?
+ 		command.config_etc.descriptor = (usb_configuration_descriptor*)fConfigurationDescriptors[i];
+ 		command.config_etc.length = tmp_config.total_length;
+ 		command.config_etc.config_index = i;
+-		if (ioctl(fRawFD, B_USB_COMMAND_GET_CONFIGURATION_DESCRIPTOR_ETC, &command, sizeof(command)) ||
++		if (ioctl(fRawFD, B_USB_RAW_COMMAND_GET_CONFIGURATION_DESCRIPTOR_ETC, &command, sizeof(command)) ||
```
Comment (pulkomandy) [inline dev-libs/libusb/patches/libusb-1.0.23.patchset]: This one is indeed a typo in the patch sent upstream and should be fixed there as well.

**Comment by pulkomandy** [comment]
No, changes were not missed. They were removed as I cleaned up the patch sent upstream. You are undoing my work, which is not acceptable.

**Comment by kenmays** [comment]
Did you look at the code release? If you think the libusb 1.0.23 code release is fine, then OK. Lets drop the PR so you can submit your approved version or the actual code release as is.

**Comment by pulkomandy** [comment]
I submitted the merge request upstream and I think I know what I'm doing, yes.

**Pull request closed without merging**